Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py

```diff
@@ -205,31 +304,259 @@ def test(client):
+
+    """
+    demisto.debug(f'PAN-DLP-Msg - {msg}')
+
+
+def update_incident(client: Client, incident_id: str, feedback: str, user_id: str, region: str,
+                    report_id: str, dlp_channel: str):
+    feedback_enum = FeedbackStatus[feedback.upper()]
+    result_json, status = client.update_dlp_incident(incident_id, feedback_enum, user_id, region, report_id, dlp_channel)
+
+    output = {
+        'feedback': feedback_enum.value,
+        'success': status == 200
+    }
+    if feedback_enum == FeedbackStatus.EXCEPTION_GRANTED:
+        minutes = result_json['expiration_duration_in_minutes']
+        output['duration'] = minutes / 60
+        result = CommandResults(
+            outputs_prefix="Exemption",
+            outputs_key_field='duration',
+            outputs=output)
+    else:
+        result = CommandResults(
+            outputs_prefix="IncidentUpdate",
+            outputs_key_field='feedback',
+            outputs=output)
+    demisto.results(result.to_context())
+
+
+def parse_incident_details(compressed_details: str):
+    details_byte_data = bz2.decompress(base64.b64decode(compressed_details))
+    details_string = details_byte_data.decode('utf-8')
+    details_obj = json.loads(details_string)
+    return details_obj
+
+
+def fetch_incidents(client: Client, regions: str, start_time: int = None, end_time: int = None):
+    if start_time and end_time:
+        print_debug_msg(f'Start fetching incidents between {start_time} and {end_time}.')
+    else:
+        print_debug_msg('Start fetching most recent incidents')
+
+    notification_map = client.get_dlp_incidents(regions=regions, start_time=start_time, end_time=end_time)
+    incidents = []
+    for region in notification_map:
+        notifications = notification_map[region]
+        for notification in notifications:
+            raw_incident = notification['incident']
+            previous_notifications = notification['previous_notifications']
+            raw_incident['region'] = region
+            raw_incident['previousNotification'] = previous_notifications[0] if len(previous_notifications) > 0 else None
+            incident_creation_time = dateparser.parse(raw_incident['createdAt'])
+            parsed_details = parse_incident_details(raw_incident['incidentDetails'])
+            raw_incident['incidentDetails'] = parsed_details
+            if not raw_incident['userId']:
+                for header in parsed_details['headers']:
+                    if header['attribute_name'] == 'username':
+                        raw_incident['userId'] = header['attribute_value']
+
+            event_dump = json.dumps(raw_incident)
+            incident = {
+                'name': f'Palo Alto Networks DLP Incident {raw_incident["incidentId"]}',
+                'type': 'Data Loss Prevention',
+                'occurred': incident_creation_time.isoformat(),  # type: ignore
+                'rawJSON': event_dump,
+                'details': event_dump
+            }
```

Comment:
Perhaps extract this code to a function - e.g. `create_incident`.

Reply:
Created a function as you suggested.